### PR TITLE
Upgrade to React Native 0.73.5

### PR DESCRIPTION
**Pull request by fernandatoledo**
### Pull request type

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Upgrade React Native version

---

### Description

The React Native version was upgraded to 0.73.5 (last stable version) using the [React Native Upgrade Helper Diff](https://react-native-community.github.io/upgrade-helper/?from=0.72.5&to=0.73.5)

---

#### Notes:

- [Flipper is no longer the debugger by default in RN 0.73.0](https://github.com/facebook/react-native/issues/43335#issuecomment-1980708164), and it will be deprecated for RN 0.74. As the new experimental debugger is still not entireley stable I still added Flipper manually, there's a known issue when used with Xcode 15.3, a recommended fix was applied from an opened issue, a comment was attached on the Podfile file to get details.

---

#### Tasks:

- [x] Tested on IOS
- [x] Tested on Android
- [ ] Tested on a small device
- [ ] Tested on a real device
- [ ] Tested all flows related with this PR changes
- [ ] Tested accessibility
- [ ] Added tests

---

### CODE REVIEW

File: rnb-cli/src/tools/react-native.ts [hunk 1]
```diff
@@ -98,11 +98,11 @@ export async function renameReactNativeApp(
     `android/app/_BUCK`,
     `android/app/BUCK`,
     `android/app/build.gradle`,
-    `android/app/src/debug/java/${newPath}/ReactNativeFlipper.java`,
+    // `android/app/src/debug/java/${newPath}/ReactNativeFlipper.java`,
```
Comment (juanchoperezj) [inline rnb-cli/src/tools/react-native.ts:101]: Do you think this comment is intended?

**Review by juanchoperezj — DISMISSED** [review]
Nice 💯

File: rnb-cli/src/tools/react-native.ts (hunk 1 above)
Comment (fernandatoledo) [reply rnb-cli/src/tools/react-native.ts]: Great! I deleted the commented line, that file is not necessary anymore on RN 0.73: https://react-native-community.github.io/upgrade-helper/?from=0.72.5&to=0.73.5

**Review by JUNIOR140889 — APPROVED** [review]

**Review by ramirogavagnin — APPROVED** [review]
Nice job ✨❤️‍🔥👏🏻

**Review by jpirazusta — APPROVED** [review]
👏 👏

**Review by juanchoperezj — APPROVED** [review]

**Pull request merged**